Quentin

Quentin-dot-prog opened the pull request:
PCA + cercle de corrélation

à voir pour le nombre de composants et si on veut retirer des colonnes de l'analyse

Quentin-dot-prog committed:
add space

Quentin-dot-prog committed:
add another space

Quentin-dot-prog committed:
Merge branch 'master' into quentin

Quentin-dot-prog committed:
PCA + cercle de corrélation

PCA + cercle de corrélation
à voir pour le nombre de composants
car avec 2 on a seulement 67 % de la variance des données

ughyper3 commented on main.py:
```diff
@@ -1,11 +1,16 @@
 from math import sqrt, log, exp
 
```
On n'importe pas de modules comme ca, il faut simplement importer les fonctions utiles, pas tout le module. Comme c'est fait par la suite

ughyper3 commented on main.py:
```diff
@@ -14,14 +19,78 @@ class Shealth:
     raw_data = read_csv('shealth.csv',  sep=";", decimal=',')
     data_set = process.process_data(raw_data)
 
+    B = data_set["day"]
```
C'est quoi "B" ?? Est ce qu'on peut avoir un nom de variable qui a du sens ?? data_set["day"] est definie plus bas en plus

ughyper3 commented on main.py:
```diff
@@ -14,14 +19,78 @@ class Shealth:
     data_set_without_categorical_variables_proj = pca.fit_transform(data_set_scale)
     variance_of_first_components = pca.explained_variance_ratio_
-
+    data_set_without_categorical_variables_proj2 = pd.DataFrame(data=data_set_without_categorical_variables_proj,
```
Pourquoi numero 2 ? tu ne peux pas modifier la variable  'data_set_without_categorical_variables_proj' qui est au dessus ?

ughyper3 commented on main.py:
```diff
@@ -14,14 +19,78 @@ class Shealth:
+                                                                columns=['PC1', 'PC2'])
+    concat_data_for_PCA = pd.concat([data_set_without_categorical_variables_proj2, B], axis=1)
+    pcs = pca.components_
+    print(data_set_without_categorical_variables.columns.tolist())
```
Le print sert a quelque chose ?

ughyper3 commented on main.py:
```diff
@@ -14,14 +19,78 @@ class Shealth:
+
+
+
+    def display_circles(pcs, n_components, pca, axis_ranks, labels=None, label_rotation=0, lims=None):
```
la methode utilise des attributs de la classe, il faut passer self en parametre de la fonction, peux tu ajouter un nom plus precis du genre display_pca_circles car on aura surement d'autres visualisation qui contiennent des cercles.
Il faut aussi une doc string qui explique les arguments et la fonction. Il faut aussi forcer les types de chaque variable et forcer le type de retour comme c'est fait sur toute les autres fonctions
Il faut aussi mettre la fonction tout en bas de la page, avec les autres methodes de classe et surtout apres tous les attributs

ughyper3 commented on main.py:
```diff
@@ -14,14 +19,78 @@ class Shealth:
+                plt.title("Cercle des corrélations des (Composants Principaux{} et {})".format(d1 + 1, d2 + 1))
+                plt.show(block=False)
+
+    display_circles(pcs, n_components, pca, [(0, 1)], labels=np.array(data_set_without_categorical_variables.columns.tolist()))
```
On appelle la methode d'affichage dans le fichier jupyter pas ici

ughyper3 commented on main.py:
```diff
@@ -14,14 +19,78 @@ class Shealth:
+
+                # nom des axes, avec le pourcentage d'inertie expliqué
+                plt.xlabel(
+                    'Composant Principal{} ({}%)'.format(d1 + 1, round(100 * pca.explained_variance_ratio_[d1], 1)))
```
On peut plutot utiliser de f-string que des formated strings

Quentin-dot-prog committed:
hist + crosstab + cramer

Quentin-dot-prog committed:
cramer "réparé je pense"

Quentin-dot-prog committed:
MDS

Quentin-dot-prog committed:
Kmeans

voir pour les valeurs à changer et à poffiner mais ça reste un modèle pour le faire pour tous les cas

ughyper3 committed:
Merge branch 'master' into quentin

# Conflicts:
#	main.py
#	visualization.ipynb

ughyper3 committed:
merge conflicts
